### DISCUSSION TITLE
Text Iwa effect - text positioning options

### DISCUSSION THREAD

**Discussion by timmolderez** [opentoonz/opentoonz#6460] [Ideas]
### Is your feature request related to a problem? Please describe

I'd like to put two pieces of text in the exact same X,Y location. (The idea is to show the first text initially, and replace it with the second text at a later point in the timeline.) More specifically: two pieces of _left-aligned_ text .. such that the beginning of each text is in the same location.

One way I tried to achieve this is with two "Text Iwa" effects. Each with a different text, but placed at the same X,Y position using the Animate tool. This works for centered text, but not for left/right-aligned text.

I can illustrate with the screenshot below. In this example, I specified a text box of 300x100mm, and set "Horizontal Align" to left. The lines of the text are indeed left-aligned, but the text in its entirity will be **centered** to that 300x100mm box. .. and because it's centered, I don't have a useful reference point that lets me put two pieces of left-aligned text in the same location.

If I could position the text in the top-left corner of that text box, that would be one way to solve it.

PS: I initially wanted to comment on [this related bug](https://github.com/opentoonz/opentoonz/issues/2463) about text alignment in the Text Iwa effect. I'd argue that this bug can be closed: the text alignment options themselves work fine actually, but because the text is always placed in the center, that can lead to confusion .. especially with only a single line of text.

![image](https://github.com/opentoonz/opentoonz/assets/3371197/803f3cf0-39b1-4672-8654-3bd0a46be51c)

### Describe the solution you'd like

I'd be happy with any solution that lets me put two pieces of text in the exact same position :) My proposal would be to add a new "Text position:" parameter to the Text Iwa effect, with options: "Center", "Top-left", "Top", "Top-right", "Right", etc.. 

Below is a quick mockup of what the different options would do:
![image](https://github.com/opentoonz/opentoonz/assets/3371197/8a7fc662-ebb7-4d22-9942-fd40a4f2f21c)

Cheers, and thank you for developing this wonderful piece of software!
, Tim

**Comment by RodneyBaker** [comment]
Hi Tim,
Have you tried the option that allows for the text to be in a Notelevel in the xsheet/timeline column?

I'll be glad to demonstrate the process if you are interested.

There are several benefits.
Perhaps the most appropriate to your case is that the position will be the same by default.

Added:  We can actually keyframe the position of the text so that might also be worth exploring.

Added:  I'm certainly in favor of Text FX Iwa being updated with more options.
Some time back I created a custom version of the dialogue which I think users would like.
I may have to recreate that as I've lost the file.

**Comment by timmolderez** [comment]
Ahh I see, that's what the "Nearby note column" is for :) That's already helpful, so I can manage with only one Text FX instead of two. The center of the text does stay in the same location, but that's not quite what I'm looking for.

Probably not necessary anymore, but I made this example .gif to help clarify what I am looking for:
The animation is flipping between two pieces of text; they're both in the exact same location .. the same center. However, I want both to be shown inside the green rectangle, such that they're aligned to the left side of that rectangle. The shorter text looks good, but the longer text not so much .. and my proposed solution could be one way to solve it.

![textfx_alignment](https://github.com/opentoonz/opentoonz/assets/3371197/58529e57-c657-4af3-aa71-3092c5e7e2a6)

In any case, many thanks for the fast response! I can certainly get by in the meantime, but thought it was a request worth sharing; would be great to have!

PS: a minor bug I noticed while playing with "Nearby note column". If I make changes to the text in my Notelevel, the changes will only show up after I force the Text FX to refresh (by changing any of its parameters).

**Comment by RodneyBaker** [comment]
It's been awhile since I did much with Text FX Iwa although I did use it just last week.
There are definitely some workarounds needed to get past some of its shortfalls.
It's a relatively new set of features so any and all feedback is appreciated.
